src/main/java/algorithms/curated170/medium/TheMaze.java

```diff
@@ -0,0 +1,66 @@
+package algorithms.curated170.medium;
+
+public class TheMaze {
+
+    public static void main(String[] args) {
+        TheMaze theMaze = new TheMaze();
+        int[][] maze = {{0, 0, 1, 0, 0},
+        {0, 0, 0, 0, 0},
+        {0, 0, 0, 1, 0},
+        {1, 1, 0, 1, 1},
+        {0, 0, 0, 0, 0}};
+
+        int[] start = {0, 4}, destination = {4, 4};
+        boolean result = theMaze.hasPath(maze, start, destination);
+        System.out.println(result);
+    }
+    boolean visited[][];
+    int[][] directions = {{0, -1}, {1, 0}, {-1, 0}, {0, 1}};
+    int[][] maze;
+    int[] destination;
+
+    public boolean hasPath(int[][] maze, int[] start, int[] destination) {
+        visited = new boolean[maze.length][maze[0].length];
+        this.maze = maze;
+        this.destination = destination;
+
+        return dps(start[0], start[1]);
+    }
+
+    public boolean dps(int x, int y) {
+
+        if (x == destination[0] && y == destination[1]) {
+            return true;
+        }
+
+        visited[x][y] = true;
+        boolean result = false;
+
+        for (int i = 0; i < directions.length; i++) {
+            int dirx = directions[i][0];
+            int diry = directions[i][1];
+            int nextX = x + dirx;
+            int nextY = y + diry;
+
+            while (canPass(nextX, nextY)
+                    && !visited[nextX][nextY]) {
+                int prx = nextX;
+                int pry = nextY;
+                nextX += dirx;
+                nextY += diry;
+                result = result || !canPass(nextX, nextY) && dps(prx, pry);
```

Comment:
I think using parenthesis here for the operators would make the code more readable.
As a note for those who do not know about the precedence of the operators. 
![image](https://user-images.githubusercontent.com/63192680/119170569-1961b480-ba6c-11eb-8b3d-66c4393aa7bf.png)
Source: http://www.math.uaa.alaska.edu/~afkjm/cs110/handouts/IfStatements.pdf

Reply:
Nice idea to add this table.